### PR TITLE
Create a Spring Boot project

### PR DESCRIPTION
**Pull request by vlykhvar**
- Create a Spring Boot project

- Added checkstyle plugin and configured Travis. 

- Added User and Role entities, the repository and service levels for them.

### CODE REVIEW

File: src/main/java/com/example/bankoflykhvar/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+@Configuration
+@ComponentScan(basePackages = {
+        "com.example.bankoflykhvar.model",
+        "com.example.bankoflykhvar.service",
+        "com.example.bankoflykhvar.repository",
+        "com.example.bankoflykhvar.initialize"
+})
+public class AppConfig {
+}
```
Comment (YuraBondarenko) [inline src/main/java/com/example/bankoflykhvar/config/AppConfig.java:14]: this class isn't necessary with spring boot

**Review by YuraBondarenko — APPROVED** [review]

File: src/main/java/com/example/bankoflykhvar/config/AppConfig.java (hunk 1 above)
Comment (vlykhvar) [reply src/main/java/com/example/bankoflykhvar/config/AppConfig.java]: agree

File: src/main/java/com/example/bankoflykhvar/model/User.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package com.example.bankoflykhvar.model;
+
+import com.sun.istack.NotNull;
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/model/User.java:3]: let's not use this annotation

File: src/main/java/com/example/bankoflykhvar/model/User.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+@Entity
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull
+    private String name;
+    @NotNull
+    private LocalDate dateOfBirth;
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/model/User.java:23]: see checklist - `If your column name contains of several words (phoneNumber) - be sure that they are correctly mapped to DB columns (phone_number).`

File: src/main/java/com/example/bankoflykhvar/model/User.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    @NotNull
+    @Column(unique = true)
+    private String phoneNumber;
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/model/User.java:26]: same

File: src/main/java/com/example/bankoflykhvar/model/User.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    @NotNull
+    private String password;
+    @OneToMany
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/model/User.java:29]: let's reconsider this relationship

File: src/main/java/com/example/bankoflykhvar/repository/RoleRepository.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.example.bankoflykhvar.repository;
+
+import com.example.bankoflykhvar.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("FROM Role WHERE roleName = ?1")
+    Optional<Role> getByName(String name);
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/repository/RoleRepository.java:10]:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

File: src/main/java/com/example/bankoflykhvar/service/UserService.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import com.example.bankoflykhvar.model.User;
+
+public interface UserService {
+    User save(User user);
+
+    User findById(Long id);
+
+    User findByPhoneNumber(String phoneNumber);
+
+    User update(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/service/UserService.java:12]: see checklist

File: src/main/java/com/example/bankoflykhvar/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
+package com.example.bankoflykhvar.service.impl;
+
+import com.example.bankoflykhvar.model.User;
+import com.example.bankoflykhvar.repository.UserRepository;
+import com.example.bankoflykhvar.service.UserService;
+import exception.EntityNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/service/impl/UserServiceImpl.java:10]: why Component?

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1,10 @@
+spring.datasource.url=jdbc:h2:file:c:\\Users\\vlykh\
+  \\IdeaProjects\\bankoflykhvar\\src\\main\\resources\\data\\data;AUTO_SERVER=true
+spring.h2.console.enabled=true
```
Comment (KseniiaMakarova) [inline src/main/resources/application.properties:3]:
```suggestion
spring.h2.console.enabled=true
spring.h2.console.path=/h2
```
by which path is the console available at the moment?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bankoflykhvar/repository/RoleRepository.java [hunk 10]
```diff
@@ -3,9 +3,7 @@
 import com.example.bankoflykhvar.model.Role;
 import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.data.jpa.repository.Query;
 
 public interface RoleRepository extends JpaRepository<Role, Long> {
-    @Query("FROM Role WHERE roleName = ?1")
-    Optional<Role> getByName(String name);
+    Optional<Role> getByRoleName(String roleName);
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/repository/RoleRepository.java:8]: does it work with parameter of type String?

File: src/main/java/com/example/bankoflykhvar/service/impl/UserServiceImpl.java [hunk 11]
```diff
@@ -33,7 +33,6 @@ public User findByPhoneNumber(String phoneNumber) {
                 -> new EntityNotFoundException("Don't find user by phoneNumber - " + phoneNumber));
     }
 
-    @Override
     public User update(User user) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/service/impl/UserServiceImpl.java:36]: let's remove this method?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bankoflykhvar/service/impl/UserServiceImpl.java [hunk 12]
```diff
@@ -34,7 +34,12 @@ public User findByPhoneNumber(String phoneNumber) {
     }
 
     public User update(User user) {
+        User newUser = findById(user.getId());
+        if (newUser == null) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/service/impl/UserServiceImpl.java:38]: can you get null here?...

File: src/main/java/com/example/bankoflykhvar/service/impl/UserServiceImpl.java [hunk 13]
```diff
@@ -34,7 +34,12 @@ public User findByPhoneNumber(String phoneNumber) {
     }
 
     public User update(User user) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankoflykhvar/service/impl/UserServiceImpl.java:36]: let's add this method on the interface level as well, if you decided so

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**